File: java/socketd/src/main/java/org/noear/socketd/transport/server/ServerProvider.java
package org.noear.socketd.transport.server;

/**
 * 服务端工厂
 *
 * @author noear
 * @since 2.0
 */
public interface ServerProvider {
    /**
     * 协议架构
     */
    String[] schemas();

    /**
     * 创建服务端
     */
    Server createServer(ServerConfig serverConfig);
}

File: javascript/socketd/src/socketd/transport/server/ServerProvider.ts
import {ServerConfig} from "./ServerConfig";
import {Server} from "./Server";

/**
 * 服务端工厂
 *
 * @author noear
 * @since 2.0
 */
export interface ServerProvider {
    /**
     * 协议架构
     */
    schemas(): string[];

    /**
     * 创建服务端
     */
    createServer(serverConfig: ServerConfig): Server;
}